GeorgiaTech/Server/Controllers/VolumeController.cs

```diff
@@ -0,0 +1,194 @@
+                    .ThenInclude(m => m.MaterialSubjects)
+                        .ThenInclude(ms => ms.MaterialSubject)
+                .Include(v => v.HomeLocation)
+                    .ThenInclude(a => a.Zip)
+                .Include(v => v.CurrentLocation)
+                    .ThenInclude(a => a.Zip);
+
+            return volumes.ToList();
+        }
+
+        /// <summary>
+        /// Find all volumes for a specific material
+        /// </summary>
+        /// <param name="materialId"></param>
+        /// <returns>A list of volumes for the specified material</returns>
+        public List<Volume> FindVolumesForMaterial(int materialId)
+        {
+            try
+            {
+                IQueryable<Volume> volumes = _context.Volumes.Where(v => v.MaterialId == materialId)
+              .Include(v => v.Material)
+                  .ThenInclude(m => m.MaterialAuthors)
+                      .ThenInclude(ma => ma.Author)
+              .Include(v => v.Material)
+                  .ThenInclude(m => m.MaterialSubjects)
+                      .ThenInclude(ms => ms.MaterialSubject)
+              .Include(v => v.HomeLocation)
+                  .ThenInclude(a => a.Zip)
+              .Include(v => v.CurrentLocation)
+                  .ThenInclude(a => a.Zip);
+
+                return volumes.ToList();
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+          
+        }
+
+        public Volume FindByID(int ID)
+        {
+            try
+            {
+                var volume = _context.Volumes
+        .Include(v => v.Material)
+            .ThenInclude(m => m.MaterialAuthors)
+                .ThenInclude(ma => ma.Author)
+        .Include(v => v.Material)
+            .ThenInclude(m => m.MaterialSubjects)
+            .ThenInclude(ms => ms.MaterialSubject)
+        .Include(v => v.HomeLocation)
+            .ThenInclude(a => a.Zip)
+         .Include(v => v.CurrentLocation)
+             .ThenInclude(a => a.Zip)
+         .Single(v => v.VolumeId == ID);
+
+                return volume;
+            }
+            catch
+            {
+                throw new NullReferenceException(message: $"Volume with id: {ID} not found");
+            }
+    
+        }
+
+        public Volume FindByType(Volume t)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Inster a volume to the database
+        /// </summary>
+        /// <param name="volume"></param>
+        /// <returns>The number of rows changed</returns>
+        public int Insert(Volume volume)
+        {
+            _context.Volumes.Add(volume);
+            return _context.SaveChanges();
+        }
+
+        /// <summary>
+        /// Updates a volume entity in the database
+        /// </summary>
+        /// <param name="volume"></param>
+        /// <returns>The number of rows changed</returns>
+        public int Update(Volume volume)
```

Comment:
Same as before: Savechanges already implements transactions, so this is largely unnecessary.